File: app/src/main/java/com/huanchengfly/tieba/api/models/SearchUserBean.java
package com.huanchengfly.tieba.api.models;

import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.huanchengfly.tieba.api.adapters.ExactMatchAdapter;
import com.huanchengfly.tieba.api.adapters.UserFuzzyMatchAdapter;
import com.huanchengfly.tieba.post.models.BaseBean;

import java.util.List;

public class SearchUserBean extends BaseBean {
    @SerializedName("no")
    private int errorCode;
    @SerializedName("error")
    private String errorMsg;
    private SearchUserDataBean data;

    public SearchUserDataBean getData() {
        return data;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public static class SearchUserDataBean {
        @SerializedName("pn")
        private int pageNum;
        @SerializedName("has_more")
        private int hasMore;
        @JsonAdapter(ExactMatchAdapter.class)
        private UserBean exactMatch;
        @JsonAdapter(UserFuzzyMatchAdapter.class)
        private List<UserBean> fuzzyMatch;

        public int getPageNum() {
            return pageNum;
        }

        public int getHasMore() {
            return hasMore;
        }

        public UserBean getExactMatch() {
            return exactMatch;
        }

        public List<UserBean> getFuzzyMatch() {
            return fuzzyMatch;
        }
    }

    public static class UserBean {
        private String id;
        private String intro;
        @SerializedName("user_nickname")
        private String userNickname;
        private String name;
        private String portrait;
        @SerializedName("fans_num")
        private String fansNum;
        @SerializedName("has_concerned")
        private int hasConcerned;

        public String getId() {
            return id;
        }

        public UserBean setId(String id) {
            this.id = id;
            return this;
        }

        public String getIntro() {
            return intro;
        }

        public UserBean setIntro(String intro) {
            this.intro = intro;
            return this;
        }

        public String getUserNickname() {
            return userNickname;
        }

        public UserBean setUserNickname(String userNickname) {
            this.userNickname = userNickname;
            return this;
        }

        public String getName() {
            return name;
        }

        public UserBean setName(String name) {
            this.name = name;
            return this;
        }

        public String getPortrait() {
            return portrait;
        }

        public UserBean setPortrait(String portrait) {
            this.portrait = portrait;
            return this;
        }

        public String getFansNum() {
            return fansNum;
        }

        public UserBean setFansNum(String fansNum) {
            this.fansNum = fansNum;
            return this;
        }

        public int getHasConcerned() {
            return hasConcerned;
        }

        public UserBean setHasConcerned(int hasConcerned) {
            this.hasConcerned = hasConcerned;
            return this;
        }
    }
}

File: app/src/main/java/com/huanchengfly/tieba/api/models/SearchUserBean.kt
package com.huanchengfly.tieba.api.models

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import com.huanchengfly.tieba.api.adapters.UserExactMatchAdapter
import com.huanchengfly.tieba.api.adapters.UserFuzzyMatchAdapter
import com.huanchengfly.tieba.post.models.BaseBean

class SearchUserBean : BaseBean() {
    @SerializedName("no")
    val errorCode: Int? = null

    @SerializedName("error")
    val errorMsg: String? = null
    val data: SearchUserDataBean? = null

    class SearchUserDataBean {
        @SerializedName("pn")
        val pageNum: Int? = null

        @SerializedName("has_more")
        val hasMore: Int? = null

        @JsonAdapter(UserExactMatchAdapter::class)
        val exactMatch: UserBean? = null

        @JsonAdapter(UserFuzzyMatchAdapter::class)
        val fuzzyMatch: List<UserBean>? = null

    }

    class UserBean {
        var id: String? = null
            private set
        var intro: String? = null
            private set

        @SerializedName("user_nickname")
        var userNickname: String? = null
            private set
        var name: String? = null
            private set
        var portrait: String? = null
            private set

        @SerializedName("fans_num")
        var fansNum: String? = null
            private set

        @SerializedName("has_concerned")
        var hasConcerned = 0
            private set

        fun setId(id: String?): UserBean {
            this.id = id
            return this
        }

        fun setIntro(intro: String?): UserBean {
            this.intro = intro
            return this
        }

        fun setUserNickname(userNickname: String?): UserBean {
            this.userNickname = userNickname
            return this
        }

        fun setName(name: String?): UserBean {
            this.name = name
            return this
        }

        fun setPortrait(portrait: String?): UserBean {
            this.portrait = portrait
            return this
        }

        fun setFansNum(fansNum: String?): UserBean {
            this.fansNum = fansNum
            return this
        }

        fun setHasConcerned(hasConcerned: Int): UserBean {
            this.hasConcerned = hasConcerned
            return this
        }
    }
}